vars/cloudFoundryCreateService.groovy

```diff
@@ -0,0 +1,176 @@
+@Field Set GENERAL_CONFIG_KEYS = STEP_CONFIG_KEYS
+@Field Set PARAMETER_KEYS = STEP_CONFIG_KEYS
+
+/**
+ * Uses the Create-Service-Push plugin to create services in a Cloud Foundry space.
+ *
+ * For details how to specify the services see the [github page of the plugin](https://github.com/dawu415/CF-CLI-Create-Service-Push-Plugin).
+ *
+ * The `--no-push` options is always used with the plugin. To deploy the application make use of the cloudFoundryDeploy step!
+ */
+@GenerateDocumentation
+void call(Map parameters = [:]) {
+    handlePipelineStepErrors (stepName: STEP_NAME, stepParameters: parameters) {
+        def script = checkScript(this, parameters) ?: this
+        def utils = parameters.juStabUtils ?: new Utils()
+        def jenkinsUtils = parameters.jenkinsUtilsStub ?: new JenkinsUtils()
+        // load default & individual configuration
+        Map config = ConfigurationHelper.newInstance(this)
+            .loadStepDefaults()
+            .mixinGeneralConfig(script.commonPipelineEnvironment, GENERAL_CONFIG_KEYS, CONFIG_KEY_COMPATIBILITY)
+            .mixinStepConfig(script.commonPipelineEnvironment, STEP_CONFIG_KEYS, CONFIG_KEY_COMPATIBILITY)
+            .mixinStageConfig(script.commonPipelineEnvironment, parameters.stageName?:env.STAGE_NAME, STEP_CONFIG_KEYS, CONFIG_KEY_COMPATIBILITY)
+            .mixin(parameters, PARAMETER_KEYS, CONFIG_KEY_COMPATIBILITY)
+            .withMandatoryProperty('cloudFoundry/org')
+            .withMandatoryProperty('cloudFoundry/space')
+            .withMandatoryProperty('cloudFoundry/credentialsId')
+            .withMandatoryProperty('cloudFoundry/serviceManifest')
+            .use()
+
+
+        utils.pushToSWA([step: STEP_NAME],config)
+
+        utils.unstashAll(config.stashContent)
+
+        if (fileExists(config.cloudFoundry.serviceManifest)) {
+            executeCreateServicePush(script, config)
+        }
+    }
+}
+
+private def executeCreateServicePush(script, Map config) {
+    dockerExecute(script:script,dockerImage: config.dockerImage, dockerWorkspace: config.dockerWorkspace) {
+
+        String varPart = varOptions(config)
+
+        String varFilePart = varFileOptions(config)
+
+        withCredentials([
+            usernamePassword(credentialsId: config.cloudFoundry.credentialsId, passwordVariable: 'CF_PASSWORD', usernameVariable: 'CF_USERNAME')
+        ]) {
+            sh """#!/bin/bash
+            set +x
+            set -e
+            export HOME=${config.dockerWorkspace}
+            cf login -u ${BashUtils.quoteAndEscape(CF_USERNAME)} -p ${BashUtils.quoteAndEscape(CF_PASSWORD)} -a ${config.cloudFoundry.apiEndpoint} -o ${BashUtils.quoteAndEscape(config.cloudFoundry.org)} -s ${BashUtils.quoteAndEscape(config.cloudFoundry.space)};
+            cf create-service-push --no-push -f ${BashUtils.quoteAndEscape(config.cloudFoundry.serviceManifest)}${varPart}${varFilePart}
+            cf logout
```

Comment:
In case something fails inside create-service-push the `logout` will never happen.

Reply:
I will change it, but this is also true for cloudFoundryDeploy btw.

Reply:
> but this is also true for cloudFoundryDeploy btw.

Agreed. Thanks. Issue #893 has been created